x-pack/plugins/case/server/services/alerts/index.ts

```diff
@@ -38,28 +38,27 @@ interface AlertsResponse {
 }
 
 export class AlertService {
-  private isInitialized = false;
-  private esClient?: IClusterClient;
-
   constructor() {}
 
-  public initialize(esClient: IClusterClient) {
-    if (this.isInitialized) {
-      throw new Error('AlertService already initialized');
-    }
-
-    this.isInitialized = true;
-    this.esClient = esClient;
-  }
-
-  public async updateAlertsStatus({ request, ids, status, index }: UpdateAlertsStatusArgs) {
-    if (!this.isInitialized) {
-      throw new Error('AlertService not initialized');
+  public async updateAlertsStatus({
+    ids,
+    status,
+    indices,
+    scopedClusterClient,
+  }: UpdateAlertsStatusArgs) {
+    /**
+     * remove empty strings from the indices, I'm not sure how likely this is but in the case that
+     * the document doesn't have _index set the security_solution code sets the value to an empty string
+     * instead
+     */
+    const sanitizedIndices = [...indices].filter((index) => index !== '');
```

Comment:
Maybe for this type of filtering we can use `isEmpty` to catch `null, undefined, etc` is case there is a change in their schema.

Reply:
Good idea will update